scipy/fft/_basic.py

```diff
@@ -1,1629 +1,358 @@
-    a little further, use `rfft`, which does the same calculation, but only
-    outputs half of the symmetrical spectrum. If the data are both real and
-    symmetrical, the `dct` can again double the efficiency, by generating
-    half of the spectrum from half of the signal.
-
-    When ``overwrite_x=True`` is specified, the memory referenced by ``x`` may
-    be used by the implementation in any way. This may include reusing the
-    memory for the result, but this is in no way guaranteed. You should not
-    rely on the contents of ``x`` after the transform as this may change in
-    future without warning.
-
-    The ``workers`` argument specifies the maximum number of parallel jobs to
-    split the FFT computation into. This will execute independent 1-D
-    FFTs within ``x``. So, ``x`` must be at least 2-D and the
-    non-transformed axes must be large enough to split into chunks. If ``x`` is
-    too small, fewer jobs may be used than requested.
-
-    References
-    ----------
-    .. [1] Cooley, James W., and John W. Tukey, 1965, "An algorithm for the
-           machine calculation of complex Fourier series," *Math. Comput.*
-           19: 297-301.
-    .. [2] Bluestein, L., 1970, "A linear filtering approach to the
-           computation of discrete Fourier transform". *IEEE Transactions on
-           Audio and Electroacoustics.* 18 (4): 451-455.
-
-    Examples
-    --------
-    >>> import scipy.fft
-    >>> import numpy as np
-    >>> scipy.fft.fft(np.exp(2j * np.pi * np.arange(8) / 8))
-    array([-2.33486982e-16+1.14423775e-17j,  8.00000000e+00-1.25557246e-15j,
-            2.33486982e-16+2.33486982e-16j,  0.00000000e+00+1.22464680e-16j,
-           -1.14423775e-17+2.33486982e-16j,  0.00000000e+00+5.20784380e-16j,
-            1.14423775e-17+1.14423775e-17j,  0.00000000e+00+1.22464680e-16j])
-
-    In this example, real input has an FFT which is Hermitian, i.e., symmetric
-    in the real part and anti-symmetric in the imaginary part:
-
-    >>> from scipy.fft import fft, fftfreq, fftshift
-    >>> import matplotlib.pyplot as plt
-    >>> t = np.arange(256)
-    >>> sp = fftshift(fft(np.sin(t)))
-    >>> freq = fftshift(fftfreq(t.shape[-1]))
-    >>> plt.plot(freq, sp.real, freq, sp.imag)
-    [<matplotlib.lines.Line2D object at 0x...>, <matplotlib.lines.Line2D object at 0x...>]
-    >>> plt.show()
-
-    """
-    return (Dispatchable(x, np.ndarray),)
+    For non-numpy arrays, this implements the Array API specification of fft.
+    For numpy arrays, see the documentation in _basic_np.py.
+    Note that if arguments outside of those in the Array API specification
+    are provided with a non-numpy array, an exception is raised.
+    """
+    if isinstance(x, np.ndarray):
+        return npbasic.fft(x, n=n, axis=axis, norm=norm, overwrite_x=overwrite_x,
+                           workers=workers, plan=plan)
+    if overwrite_x is not False:
+        Exception
```

Comment:
What we want to do here is raise an error via `raise SomeError('...some comment')`. I think ValueError is the appropriate one, there should be an example somewhere else in SciPy of a similar error we can use.